sky/spot/spot_utils.py

```diff
@@ -68,6 +68,41 @@ def get_job_status(backend: 'backends.CloudVmRayBackend',
     return status
 
 
+def update_spot_job_status(job_id: Optional[int] = None):
+    """Update spot job status if the controller failed abnormally.
```

Comment:
nit: if the controller [process]

We should distinguish controller (VM) from controller processes when possible. Same on L74?